db.go

```diff
@@ -0,0 +1,27 @@
+package db
+
+// Generalized PIR database interface to be implemented
+// TODO: Determine concrete function signatures for this interface
+type PIRDB interface {
+	// Initialized the PIR database with some data
+	Init() error
+	// Updates the PIR database with new data
+	Update() error
```

Comment:
I'm not 100% sure about this, but a replace operation may be much easier to implement and reason about than update the index when the PIR system has an offline stage.  The pre-computing offline stages are used for improving query performance.

Reply:
What if you need to update the database as in add more data to it? Perhaps, I can call it `AddData()` instead of `Update()`. Ideally, you don't want to be replacing the entire database all the time since that is space inefficient for many use cases.

Reply:
That's a good point. I meant the `Update()` to be atomic wrt to an entry in the database. So we could:

```
 Update(*Item{A}, 100) // this performs an Upsert on index 100 with the item{A} 
```

So I think we're on the same page here.

Reply:
I think that it'll do.